### ISSUE TITLE
We need a pre-check to ensure that NOR requests are checked for illegal URI schemes

### ISSUE THREAD

**Issue by matentzn** [OBOFoundry/OBO-Dashboard#91]
ROBOT report has:

https://github.com/ontodev/robot/blob/master/robot-core/src/main/resources/report_queries/invalid_entity_uri.rq

https://github.com/ontodev/robot/blob/master/robot-core/src/main/resources/report_profile.txt#L32

However, when someone (see e.g. https://github.com/OBOFoundry/OBOFoundry.github.io/issues/2371) creates completely malformed URIs (https, and different ID scheme), the automatic base extractor misses _all_ classes (e.g. no class is tested in the Report). 

We need some efficient way to run the above checks _before_ the base extraction.

**Comment by bpeters42** [comment]
My stab at this would be to do a pre-check for each term URI if it contains "obolibrary.org". And if it does, check if it conforms with the overall format "http://purl.obolibrary.org/obo/XXX". Of course this wouldn't catch instances of misspelled obolibrary.org. And there might be a weird case where such a URI is not intended to be an obolibray term - but in those cases I would question if there isn't another problem (I don't think we want to encourage URIs like HTTP://better_than_obolibray.org' or 'google.com?obolibray.org)...

**Comment by pfabry** [comment]
In my opinion, these checks could initially be carried out manually by the NOR manager until sufficiently reliable automatic checks are available. Right now, I've followed the NOR manager SOP to the letter (maybe too much in fact) that is to configure the NOR dashboard for new ontologies and, when everything is green, pass it to the reviewer. 

I feel there are additional tasks that could be done prior to the review, tasks that would have to be done manually as a first step. The illegal URIs is one, but there are also problems with the object properties cf. https://github.com/OBOFoundry/obo-nor.github.io/issues/52
Regarding GNDO as an example, the fact that it is a taxonomy without any "transversal" axioms using object properties or the fact that BFO:role is extensively used without specifying bearers or realizing processes should raise questions for the authors before even going to the review.

Maybe we could split the review process into form and content:

- form would be the actual NOR review + additional checks (URIs, Object properties, Imports, release artefacts, etc.)
- content would be focused on the domain knowledge the ontology is about and how it fits with the OBO foundry

**Comment by matentzn** [comment]
> In my opinion, these checks could initially be carried out manually by the NOR manager until sufficiently reliable automatic checks are available. 

I agree, I made a [PR](https://github.com/OBOFoundry/OBOFoundry.github.io/pull/2394) to that end, but I put this in the ballpark of the main reviewer. Can you make an alternative PR with your NOR plus review?

**Comment by pfabry** [comment]
Will do. 
In addition, there are actually 2 places where the NOR manager SOP are documented: in `docs\SOP.md` and in `docs\roles\RoleNORManager.md`, the latter being much more detailed.
I'd propose to give an overview of the NOR Manager role in `SOP.md` and add a link to `RoleNORManager.md` where the SOP are fully decribed.

**Comment by matentzn** [comment]
Agreed!
